perllib/FixMyStreet/DB/ResultSet/DefectType.pm

```diff
@@ -0,0 +1,22 @@
+package FixMyStreet::DB::ResultSet::DefectType;
+use base 'DBIx::Class::ResultSet';
+
+use strict;
+use warnings;
+
+sub for_bodies {
+    my ($rs, $bodies, $category) = @_;
+    my $attrs = {
+        'me.body_id' => $bodies,
+    };
+    if ($category) {
+        $attrs->{'contact.category'} = [ $category, undef ];
+    }
+    $rs->search($attrs, {
+        order_by => 'name',
+        join => { 'contact_defect_types' => 'contact' },
```

Comment:
I think this function is always called with a category, so doesn't really matter, but we don't want to do this join if $category is not set, I imagine.